test/integration/dryrun/dryrun_test.go

```diff
@@ -0,0 +1,677 @@
+	patch := []byte(`{"metadata":{"annotations":{"patch": "true"}}}`)
+	obj, err := rsc.Patch(name, types.MergePatchType, patch, metav1.UpdateOptions{DryRun: []string{metav1.DryRunAll}})
+	if err != nil {
+		t.Fatalf("failed to dry-run patch object: %v", err)
+	}
+	if _, ok := obj.GetAnnotations()["patch"]; !ok {
+		t.Fatalf("dry-run patched annotations should be returned, got: %v", obj.GetAnnotations())
+	}
+	obj, err = rsc.Get(obj.GetName(), metav1.GetOptions{})
+	if err != nil {
+		t.Fatalf("failed to get object: %v", err)
+	}
+	if _, ok := obj.GetAnnotations()["patch"]; ok {
+		t.Fatalf("dry-run patched annotations should not be persisted, got: %v", obj.GetAnnotations())
+	}
+}
+
+func DryRunStatusUpdateTest(t *testing.T, rsc dynamic.ResourceInterface, name string) {
+	obj, err := rsc.Get(name, metav1.GetOptions{}, "status")
+	if err == nil {
+		_, err = rsc.Update(obj, metav1.UpdateOptions{DryRun: []string{metav1.DryRunAll}})
+		if err != nil {
+			t.Fatalf("dry-run update on status resource failed: %v", err)
+		}
+	}
+}
+
+func DryRunUpdateTest(t *testing.T, rsc dynamic.ResourceInterface, name string) {
+	obj, err := rsc.Get(name, metav1.GetOptions{})
+	if err != nil {
+		t.Fatalf("failed to retrieve object: %v", err)
+	}
+	obj.SetAnnotations(map[string]string{"update": "true"})
+	obj, err = rsc.Update(obj, metav1.UpdateOptions{DryRun: []string{metav1.DryRunAll}})
+	if err != nil {
+		t.Fatalf("failed to dry-run update object: %v", err)
+	}
+	if _, ok := obj.GetAnnotations()["update"]; !ok {
+		t.Fatalf("dry-run updated annotations should be returned, got: %v", obj.GetAnnotations())
+	}
+
+	obj, err = rsc.Get(obj.GetName(), metav1.GetOptions{})
+	if err != nil {
+		t.Fatalf("failed to get object: %v", err)
+	}
+	if _, ok := obj.GetAnnotations()["update"]; ok {
+		t.Fatalf("dry-run updated annotations should not be persisted, got: %v", obj.GetAnnotations())
+	}
+}
+
+func DryRunDeleteCollectionTest(t *testing.T, rsc dynamic.ResourceInterface, name string) {
+	err := rsc.DeleteCollection(&metav1.DeleteOptions{DryRun: []string{metav1.DryRunAll}}, metav1.ListOptions{})
+	if err != nil {
+		t.Fatalf("dry-run delete collection failed: %v", err)
+	}
+	obj, err := rsc.Get(name, metav1.GetOptions{})
+	if err != nil {
+		t.Fatalf("failed to get object: %v", err)
+	}
+	ts := obj.GetDeletionTimestamp()
+	if ts != nil {
+		t.Fatalf("Object has a deletion timestamp after dry-run delete collection")
+	}
+}
+
+func DryRunDeleteTest(t *testing.T, rsc dynamic.ResourceInterface, name string) {
+	err := rsc.Delete(name, &metav1.DeleteOptions{DryRun: []string{metav1.DryRunAll}})
+	if err != nil {
+		t.Fatalf("dry-run delete failed: %v", err)
+	}
+	obj, err := rsc.Get(name, metav1.GetOptions{})
+	if err != nil {
+		t.Fatalf("failed to get object: %v", err)
+	}
+	ts := obj.GetDeletionTimestamp()
+	if ts != nil {
+		t.Fatalf("Object has a deletion timestamp after dry-run delete collection")
```

Comment:
Not important, but i think it should be "Object has a deletion timestamp after dry-run delete collection"

Reply:
I assumed you meant to remove "collection" ;-)